src/callbacks_step/amr_dg2d.jl

```diff
@@ -1,3 +1,95 @@
+# Redistribute data for load balancing after partitioning the mesh
+function redistribute_data!(u_ode::AbstractVector, mesh::TreeMesh{2}, equations, 
```

Comment:
I don't want to be nitpicky, but I don't like the name `redistribute_data!`, as it suggests (at least to me) that it really is mainly a communication routine (and thus comparably inexpensive, save for communication overhead). However, we also tear down and recreate all data structures, which is potentially quite costly. Thus I'd suggest to use a more descriptive or a more general name such as `rebalance_solver!` (feel free to suggest alternatives that are better, I am not entirely happy with it either).